src/middlewared/middlewared/plugins/crypto_/dhparams.py

```diff
@@ -11,15 +13,15 @@ class Config:
 
     @private
     async def dhparam(self):
-        return '/data/dhparam.pem'
+        return DHPARAM_PEM_PATH
+
 
     @private
     @job()
     def dhparam_setup(self, job):
-        dhparam_path = self.middleware.call_sync('certificate.dhparam')
-        if not os.path.exists(dhparam_path) or os.stat(dhparam_path).st_size == 0:
-            with open('/dev/console', 'wb') as console:
-                with open(dhparam_path, 'wb') as f:
-                    subprocess.run(
-                        ['openssl', 'dhparam', '-rand', '/dev/urandom', '2048'], stdout=f, stderr=console, check=True
-                    )
+        """Generate dhparam.pem if it doesn't exist, or has no data in it"""
+        with open(DHPARAM_PEM_PATH, 'a+') as f:
+            if os.fstat(f.fileno()).st_size == 0:
```

Comment:
Should we validate / set appropriate permissions on file?

Reply:
Yes please..i am already pushing a PR which fixes perms stuff in `/data` and this is one of those files. It should have `0o600`

Reply:
I didn't do perm validation on my end because of the PR from @Qubad786 . Currently the perms are `0o644` in the tests, once that gets merged it should be the proper `0o600`!

Reply:
I believe they need to be looser than 0o600 to facility key exchange.

Reply:
I was getting ready to say that we should probably validate whether or not 0600 is too restrictive

Reply:
There aren't any issues when i use `0o600` and it being owned by root..so nginx master process which runs as root reads this which is why we don't have any problems. I have tested https and certs creation etc and we should be good ✅ 
ref: https://superuser.com/a/1250447